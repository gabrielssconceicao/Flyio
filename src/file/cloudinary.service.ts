import { BadRequestException, Injectable } from '@nestjs/common';
import { v2 as cloudinary } from 'cloudinary';
import { Readable } from 'stream';
@Injectable()
export class CloudinaryService {
  constructor() {
    cloudinary.config({
      cloud_name: process.env.CLOUDINARY_CLOUD_NAME,
      api_key: process.env.CLOUDINARY_API_KEY,
      api_secret: process.env.CLOUDINARY_API_SECRET,
    });
  }

  private renameFile(filename: string): string {
    const extension = filename.split('.').pop();
    const timestamp = Date.now();
    return `file_${timestamp}.${extension}`;
  }

  async uploadProfilePicture(file: Express.Multer.File): Promise<string> {
    const { originalname, buffer } = file;

    // Realizar as verificações de tamanho e tipo de mime

    const renamedFile = this.renameFile(originalname);

    return new Promise((resolve, reject) => {
      try {
        // Create a Readable stream from the buffer
        const bufferStream = Readable.from(buffer);

        // Upload config to Cloudinary
        const uploadStream = cloudinary.uploader.upload_stream(
          {
            resource_type: 'image',
            folder: 'profile-pictures',
            public_id: renamedFile,
          },
          (error, result) => {
            if (error) {
              reject(
                new BadRequestException('Error uploading profile picture.'),
              );
            } else {
              // Retornar a URL segura da imagem após o upload
              resolve(result.secure_url);
            }
          },
        );

        // Passar o stream para o upload
        bufferStream.pipe(uploadStream);
      } catch {
        reject(new BadRequestException('Error uploading profile picture.'));
      }
    });
  }

  async uploadPostImage(file: Express.Multer.File): Promise<string> {
    const { mimetype, size, filename } = file;

    // this.checkSize(size, 10 * 1024, 15 * 1024 * 1024);
    // this.validateMimeType(mimetype, ['image/jpeg', 'image/png', 'image/gif']);

    return `Post image uploaded: ${filename}`;
  }
}
