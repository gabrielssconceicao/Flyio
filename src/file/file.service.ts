import { BadRequestException, Injectable } from '@nestjs/common';

@Injectable()
export class FileService {
  private throwBadRequest(msg: string): void {
    throw new BadRequestException(msg);
  }

  private checkSize(size: number, minSize: number, maxSize: number): void {
    if (size < minSize)
      this.throwBadRequest(
        `File too small. Minimum size: ${minSize / 1024} KB`,
      );
    if (size > maxSize)
      this.throwBadRequest(
        `File too large. Maximum size: ${maxSize / (1024 * 1024)} MB`,
      );
  }

  private validateMimeType(mimetype: string, allowedTypes: string[]): void {
    if (!allowedTypes.includes(mimetype)) {
      this.throwBadRequest(
        `Invalid file type. Allowed types: ${allowedTypes.join(', ')}`,
      );
    }
  }

  async uploadProfilePicture(file: Express.Multer.File): Promise<string> {
    const { mimetype, size, filename } = file;

    this.checkSize(size, 10 * 1024, 5 * 1024 * 1024);
    this.validateMimeType(mimetype, ['image/jpeg', 'image/png']);

    // Simulação de uploa
    return `Profile picture uploaded: ${filename}`;
  }

  async uploadPostImage(file: Express.Multer.File): Promise<string> {
    const { mimetype, size, filename } = file;

    this.checkSize(size, 10 * 1024, 15 * 1024 * 1024);
    this.validateMimeType(mimetype, ['image/jpeg', 'image/png', 'image/gif']);

    return `Post image uploaded: ${filename}`;
  }
}
