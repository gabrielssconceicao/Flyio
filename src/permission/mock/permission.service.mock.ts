export const permissionServiceMock = {
  verifyUserOwnership: jest.fn(),
};
