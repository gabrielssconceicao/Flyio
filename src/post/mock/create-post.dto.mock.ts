import { CreatePostDto } from '../dto';
export const generateCreatePostDtoMock = (): CreatePostDto => {
  return {
    text: 'Hello world',
  };
};
