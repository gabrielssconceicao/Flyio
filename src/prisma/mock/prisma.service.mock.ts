export const userPrismaService = {
  user: {
    create: jest.fn(),
    findFirst: jest.fn(),
    findMany: jest.fn(),
    findUnique: jest.fn(),
    update: jest.fn(),
    delete: jest.fn(),
    count: jest.fn(),
  },
};

export const postPrismaService = {
  post: {
    create: jest.fn(),
    findMany: jest.fn(),
    count: jest.fn(),
    findUnique: jest.fn(),
    delete: jest.fn(),
  },
  postLikes: {
    create: jest.fn(),
    delete: jest.fn(),
  },
};

export const selectUserFieldsMock = {
  id: true,
  name: true,
  username: true,
  profileImg: true,
  bio: true,
  active: true,
  _count: { select: { following: true, followers: true } },
};
