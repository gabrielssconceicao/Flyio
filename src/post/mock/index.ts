import { postMock } from './post.mock';
import { generateCreatePostDtoMock } from './create-post.dto.mock';
export { postMock, generateCreatePostDtoMock };
