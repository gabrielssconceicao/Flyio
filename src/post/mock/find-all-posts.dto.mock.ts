import { FindAllPostsResponseDto } from '../dto/find-all-posts.dto';
import { generatedPostMock } from './post.mock';

export function generateFindAllPostsDtoMock(): FindAllPostsResponseDto {
  return {
    count: 1,
    items: [generatedPostMock()],
  };
}
