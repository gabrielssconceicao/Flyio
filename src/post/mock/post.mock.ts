import { PostEntity } from '../entities/post.entity';

export const postMock: PostEntity = {
  id: '42-d-f-df4',
  content: 'Hello world',
  createdAt: new Date(),
  user: {
    username: 'jDoe453',
    profileImg: 'https://example.com/profile.jpg',
  },
  images: [],
};
