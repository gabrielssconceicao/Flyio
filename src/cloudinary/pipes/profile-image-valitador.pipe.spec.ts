import { BadRequestException } from '@nestjs/common';
import { generateFileMock } from '../mock/file.mock';
import { ProfileImageValidatorPipe } from './profile-image-validator.pipe';

describe('<ProfileImageValidatorPipe/ >', () => {
  let pipe: ProfileImageValidatorPipe;
  let file: Express.Multer.File;
  let mimetype: string[];
  beforeEach(() => {
    mimetype = ['image/jpeg', 'image/png'];
    pipe = new ProfileImageValidatorPipe();
    file = generateFileMock();
  });

  it('should be defined', () => {
    expect(pipe).toBeDefined();
  });
  it('should return the file', () => {
    const result = pipe.transform(file, {} as any);
    expect(result).toEqual(file);
    expect(result).toMatchSnapshot();
  });
  it('should return null if the file is null', () => {
    const result = pipe.transform(null, {} as any);
    expect(result).toBeNull();
    expect(result).toMatchSnapshot();
  });
  it('should throw an error if the file is too small', () => {
    file.size = 9 * 1024;
    expect(() => pipe.transform(file, {} as any)).toThrow(BadRequestException);
    expect(() => pipe.transform(file, {} as any)).toThrow(
      'File too small. Minimum size: 10 KB',
    );
  });
  it('should throw an error if the file is too big', () => {
    file.size = 6 * 1024 * 1024;
    expect(() => pipe.transform(file, {} as any)).toThrow(BadRequestException);
    expect(() => pipe.transform(file, {} as any)).toThrow(
      'File too large. Maximum size: 5 MB',
    );
  });
  it('should throw an error if the file type is not allowed', () => {
    file.mimetype = 'application/pdf';
    expect(() => pipe.transform(file, {} as any)).toThrow(BadRequestException);
    expect(() => pipe.transform(file, {} as any)).toThrow(
      `Invalid file type. Allowed types: ${mimetype.join(', ')}`,
    );
  });
});
