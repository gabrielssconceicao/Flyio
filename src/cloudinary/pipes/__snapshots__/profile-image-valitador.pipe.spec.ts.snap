// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProfileImageValidatorPipe/ > should return null if the file is null 1`] = `null`;

exports[`<ProfileImageValidatorPipe/ > should return the file 1`] = `
{
  "buffer": {
    "data": [
      116,
      101,
      115,
      116,
    ],
    "type": "Buffer",
  },
  "mimetype": "image/jpeg",
  "originalname": "image.jpg",
  "size": 10240,
}
`;
